utils/DeployUtils.ps1

```diff
@@ -161,63 +161,141 @@ function ConfigureScubaGearModule{
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyCollection()]
+        [array]
+        $LiteralFilePaths = @(),
+        [Parameter(Mandatory=$false)]
+        [AllowEmptyCollection()]
+        [array]
+        $Extensions = @()
+    )
+
+    $FileNames = @()
+
+    if ($Extensions.Count -gt 0){
+        $FileNames += Get-ChildItem -Recurse -Path $SourcePath -Include $Extensions
+    }
+    
+    Write-Debug "Found $($FileNames.Count) files to sign" 
+
+    $FileList = New-TemporaryFile
+    $FileNames.FullName | Out-File -FilePath $($FileList.FullName) -Encoding utf8 -Force
+    Write-Debug "Files: $(Get-Content $FileList)"
+    return $FileList.FullName
+}
+
+function CallAzureSignTool{
+    <#
+    .NOTES
+    Internal function 
     #>
     param (
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({[uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'https'})]
+        [System.Uri]
+        $AzureKeyVaultUrl,
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $CertificateName,
         [Parameter(Mandatory=$true)]
         [ValidateScript({Test-Path -Path $_})]
         $ModulePath,
         [Parameter(Mandatory=$false)]
-        [scriptblock]
-        $GetCertificate = $function:GetCodeSigningCertificate,
-        [Parameter(Mandatory=$false)]
         [ValidateScript({[uri]::IsWellFormedUriString($_, 'Absolute') -and ([uri] $_).Scheme -in 'http','https'})]
         $TimeStampServer = 'http://timestamp.digicert.com',
-        [Parameter(Mandatory=$false)]
-        [ValidateSet('SHA256')]
-        [string]
-        $HashAlgorithm = 'SHA256'
+        [Parameter(Mandatory=$true)]
+        [ValidateScript({Test-Path -Path $_ -PathType Leaf})]
+        $FileList
     )
-    #TODO: Add code signing
-    $CatalogFileName = 'ScubaGear.cat'
-    $CatalogPath = Join-Path -Path $ModulePath -ChildPath $CatalogFileName
-    $Cert = Invoke-Command $GetCertificate
 
-    # Digitally sign scripts, manifest, and modules
-    Get-ChildItem $ModulePath -Include *.psd1,*psm1,*.ps1 -Recurse |
-    Set-AuthenticodeSignature -Certificate $Cert -TimeStampServer $TimeStampServer -HashAlgorithm $HashAlgorithm
+    $SignArguments = @(
+        'sign',
+        '-coe',
+        '-fd',"sha256",
+        '-tr', $TimeStampServer,
+        '-kvu',$AzureKeyVaultUrl,
+        '-kvc',$CertificateName,
+        '-kvm'
+        '-ifl',$FileList         
+    )
 
-    New-FileCatalog -Path $ModulePath -CatalogFilePath $CatalogPath -CatalogVersion 2.0 -Verbose
-    Set-AuthenticodeSignature -FilePath $CatalogPath -Certificate $Cert -TimestampServer $TimeStampServer -HashAlgorithm $HashAlgorithm
+    Write-Debug "Calling AzureSignTool: $SignArguments"
 
-    $TestResult = Test-FileCatalog -CatalogFilePath $CatalogPath
-    return 'Valid' -eq $TestResult.Status
+    $ToolPath = (Get-Command AzureSignTool).Path  
+    powershell -Command "& $ToolPath $SignArguments"      
```

Comment:
This is a little strange, to launch a powershell subprocess.  Can you just use

& $ToolPath  @SignArguments

?

Or, if you need a subexpression, something like:

$(& $ToolPath @SignArguments)

That is how we invoke rego.

Reply:
I do not have a rationale to deviate from OPA approach.  I will change to suggested.